src/mdast-process-images.js

```diff
@@ -12,28 +12,78 @@
 import { visit } from 'unist-util-visit';
 import processQueue from '@adobe/helix-shared-process-queue';
 
+function createFilter(log, baseUrlStr, imgSrcPolicy) {
+  const baseUrl = new URL(baseUrlStr);
+  const imgSrcPolicFilters = imgSrcPolicy.map((policyValue) => {
+    // filter according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/Sources
+    if (policyValue === '*') {
+      return () => true;
+    }
+
+    if (policyValue === 'self') {
+      return (urlStr) => {
+        try {
+          const { host } = new URL(urlStr);
+          return baseUrl.host === host;
+        } catch (e) {
+          log.warn(`Failed to parse url '${urlStr}': ${e.message}`);
+          return false;
+        }
+      };
+    }
+
+    return (urlStr) => {
+      try {
+        // omit the protocol check, as https:// is enforced earlier already
+        // omit the pathname check, as we may not need it (yet)
+        const url = new URL(urlStr);
```

Comment:
same as as above: 

check for urls starting with `/`

```suggestion
        const url = new URL(urlStr, baseUrl);
```